extensions/standard-processors/processors/SplitContent.cpp

```diff
@@ -0,0 +1,168 @@
+  if (!next_split) {
+    throw Exception(PROCESSOR_EXCEPTION, "Couldn't create FlowFile");
+  }
+  if (keep_byte_sequence && byte_sequence_location_ == ByteSequenceLocation::Leading) {
+    session.appendBuffer(next_split, byte_sequence_);
+  }
+  return next_split;
+}
+
+void SplitContent::finalizeLatestSplitContent(core::ProcessSession& session, const std::shared_ptr<core::FlowFile>& latest_split, const std::vector<std::byte>& buffer) const {
+  const std::span<const std::byte> data_without_byte_sequence{buffer.data(), buffer.size() - byte_sequence_.size()};
+  session.appendBuffer(latest_split, data_without_byte_sequence);
+  if (keep_byte_sequence && byte_sequence_location_ == ByteSequenceLocation::Trailing) {
+    session.appendBuffer(latest_split, byte_sequence_);
+  }
+}
+
+void SplitContent::finalizeLastSplitContent(core::ProcessSession& session,
+    std::vector<std::shared_ptr<core::FlowFile>>& splits,
+    const std::vector<std::byte>& buffer,
+    const bool ended_with_byte_sequence) const {
+  if (ended_with_byte_sequence && splits.back()->getSize() != 0) {
+    if (keep_byte_sequence && byte_sequence_location_ == ByteSequenceLocation::Leading) {
+      const auto last_split = session.create();
+      if (!last_split) {
+        throw Exception(PROCESSOR_EXCEPTION, "Couldn't create FlowFile");
+      }
+      splits.push_back(last_split);
+      session.appendBuffer(splits.back(), byte_sequence_);
+    }
+  } else {
+    session.appendBuffer(splits.back(), buffer);
+  }
+}
+
+namespace {
+std::shared_ptr<core::FlowFile> createFirstSplit(core::ProcessSession& session) {
+  auto first_split = session.create();
+  if (!first_split) {
+    throw Exception(PROCESSOR_EXCEPTION, "Couldn't create FlowFile");
+  }
+  return first_split;
+}
+
+void updateSplitAttributesAndTransfer(core::ProcessSession& session, const std::vector<std::shared_ptr<core::FlowFile>>& splits, const core::FlowFile& original) {
+  const std::string fragment_identifier_ = utils::IdGenerator::getIdGenerator()->generate().to_string();
+  for (size_t split_i = 0; split_i < splits.size(); ++split_i) {
+    const auto& split = splits[split_i];
+    split->setAttribute(SplitContent::FragmentCountOutputAttribute.name, std::to_string(splits.size()));
+    split->setAttribute(SplitContent::FragmentIndexOutputAttribute.name, std::to_string(split_i + 1));  // One based indexing
+    split->setAttribute(SplitContent::FragmentIdentifierOutputAttribute.name, fragment_identifier_);
+    split->setAttribute(SplitContent::SegmentOriginalFilenameOutputAttribute.name, original.getAttribute(core::SpecialFlowAttribute::FILENAME).value_or(""));
+    session.transfer(split, SplitContent::Splits);
+  }
+}
+
+bool lastSplitIsEmpty(const std::vector<std::shared_ptr<core::FlowFile>>& splits) {
+  return splits.back()->getSize() != 0;
+}
+}  // namespace
+
+void SplitContent::onTrigger(core::ProcessContext& context, core::ProcessSession& session) {
+  gsl_Assert(!byte_sequence_.empty());
```

Comment:
It was ensured implicitly because byte_sequence was a required property so it would have thrown if it was empty, but I've added an explicit check in onschedule aswell